### ISSUE TITLE
Inventory control input UX improvement

### ISSUE THREAD

**Issue by mezanddav** [overextended/ox_inventory#1765]
I'd like an improvement to the inventory amount input behaviour. We got a lot of complaints about how it works.
Issues:
- Input can't be empty. This confuses everyone because I saw them manually enter "01" go back and delete the "0" from the start. It's not a good UX.
- Players are having difficulty moving full stacks. They either type in the exact amount or something lower. This is a massive problem when it comes to money for example. If they enter a large amount - They get error and will not move whatever amount is available.

**Solution**
- Allow input to be empty and in case of invalid (empty) input on focus loss reset to the previous amount.
- When the amount is "0" - select full stack instead of 1 item.

**It's so annoying even as an admin, that I considered doing it myself. But this could be a small change to the codebase but a massive improvement to the script.**

**Comment by thelindat** [comment]
> I saw them manually enter "01" go back and delete the "0" from the start.

The input box already formats the text as an integer - if it has 0 and you type 1, it will be 1.

> If they enter a large amount - They get error

[I don't see any error here](https://github.com/user-attachments/assets/1c0663e3-c059-4a7f-a4f3-6dfaaaeec19b). I can also input "10000000000000000000" or even "1.00000000009e+21" without any issue.

> When the amount is "0" - select full stack instead of 1 item.

0 already _is_ equal to moving a full stack.

**Comment by mezanddav** [comment]
Thank you for your quick answer. 

> The input box already formats the text as an integer - if it has 0 and you type 1, it will be 1.
We can't expect users to know this. It's also not ideal UX. It should allow for empty input on focus.

> If they enter a large amount - They get an error
Sorry, my mistake for not explaining it properly. Let's say they have $1.423.532 in their inventory. It's a hard and cumbersome number. If they enter $1.500.00 and try to move the item, they get an error that they don't have enough items to move. It should just move what is available. But this is irrelevant if the above is true:

> It already can be..
Just tested it. The issue is that I had to press backspace 3 times before it was deleted. We can't expect players to know this either - I didn't notice either. 

> When the amount is "0" - select full stack instead of 1 item.
We tested this with "Give" action and it doesn't work.

**Comment by thelindat** [comment]
> We can't expect users to know this.

_You're_ the one saying they typing "01" when it can't be "01". You're complaining about something that does not happen.

>  If they enter $1.500.00 and try to move the item, they get an error that they don't have enough items to move.

No, they don't. Hence me saying there was no issue with me entering "10000000000000000000", when I only have 5000 of the item. You're again complaining about something that does not happen.

> We tested this with "Give" action and it doesn't work.

Again, you're complaining about something that is not the case - you can even see here.
https://github.com/overextended/ox_inventory/blob/main/modules/inventory/server.lua#L2410


---

Basically, you're using some unsupported version of ox_inventory which has a number of problems which are not part of the official resource.

**Comment by thelindat** [comment]
> an error that they don't have enough items to move

A _notification_ is not an error. Furthermore, the only notification that exists about "not enough items" is when trying to _use_ an item which you do not have enough of. There is _never_ a message about not having enough items to _move_ - because trying to move "too many" or "0" items will simply move _the entire stack_.

**Comment by mezanddav** [comment]
Why do developers often become defensive and dismissive when support is requested? I clearly outlined the issues with the latest build of the addon, yet you continue to deny them. It's fine; we'll address it on our server. Thank you.

**Comment by LukeWasTakenn** [comment]
The amount input field works fine, if you feel otherwise the resource is open source and you're welcome to contribute your solution and we'll consider it.

**Comment by thelindat** [comment]
> I clearly outlined the issues with the latest build of the addon, yet you continue to deny them. It's fine; we'll address it on our server. Thank you.

These issues you mentioned simply _do not exist_ with the resource as it currently exists, nor have they been parts of the resource previously. I don't know what you want me to do when you're the _only person_ experiencing this problem. We aren't "denying" them, we are simply pointing out that your version of ox_inventory very clearly is not _ours_.

> Why do developers often become defensive and dismissive when support is requested?

Why are FiveM server owners so aggressive, dismissive, and entitled when "requesting" (see: demanding) support?

I'm dismissing these issues because _they are not present_. How is it "defensive" to point out that the resource you are running and the resource we are maintaining are very clearly different? How are we supposed to address _your_ problems? You didn't even bother to fill out a form with important information, which I generally just ignore and close immediately anyway.
